hbase-common/src/main/java/org/apache/hadoop/hbase/io/encoding/EncodingState.java

```diff
@@ -31,4 +32,15 @@
    * The previous Cell the encoder encoded.
    */
   protected Cell prevCell = null;
+
+  @VisibleForTesting
+  public Cell getLastCell() {
+    return prevCell;
+  }
+
+  public void beforeShipped() {
+    if (this.prevCell != null) {
+      this.prevCell = KeyValueUtil.copyToNewKeyValue(this.prevCell);
```

Comment:
Seems we need both key and value from the prevCell in FastDiffDeltaEncoder and all.  Just add a comment saying so. Normally we copy the key part alone in beforeShipped

Reply:
done it